platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/MigrationEntry.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * The <code>MigrationEntry</code> interfaces provides support for artifacts that are being exported
+ * or imported during migration.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface MigrationEntry extends Comparable<MigrationEntry> {
+    /**
+     * Gets the migration report associated with this entry. Warnings or errors can be recorded with
+     * the report.
```

Comment:
An entry is just something exported as a blob nothing more. They are identified using a path. The property referenced entries are an artifact that allows one to get the entry that correspond to the value stored for a given property when properties are used in the system to referenced files.